doc/data_io_pipeline_design.md

```diff
@@ -0,0 +1,20 @@
+# Data IO Pipeline Design
+
+Data IO pipeline for ElasticDL involves reading data from [RecordIO](https://github.com/elasticdl/recordio) file or ODPS table, making data generator for tf.data.Dataset and parsing features and labels by dataset_fn user defined (see <em> Figure 1</em>).
+
+<center>
+    <img src="figures/data_io_pipeline.jpg" height="400" width="450">
+    <br>
+    <div style="
+    display: inline-block;
+    color: #999;
+    padding: 2px;"><em>Figure 1 </em>. ElasticDL data IO pipeline</div>
+</center>
```

Comment:
Better to use `markdown` tags here. For the title, I guess we can skip it: `see the following figure`.

Reply:
Good idea, updated.